docs/events.md

```diff
@@ -0,0 +1,47 @@
+# Player Events
+The player event system uses an event target like API to register, unregister and dispatch events.
+
+## Overview
+
+The player events are consisted of two event types:
+
+ 1. HTML5 events - various events sent by the browser when handling media embedded using the `<video>` element. The player runs on top of HTML video element, which may trigger the events. A reference to those events can be found [here](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events).
+
+ 2. Player custom events - special events indicating changes in the state of the player that does not exist in the html5 video event list and are related to the integral behavior of the player. Such as ads, fullscreen and subtitles events.
+
+Only DOM elements may be true EventTargets. Inside the player, we use fake events targets (fake-event-target.js), and fake events (fake-event) in order to dispatch events not from a DOM element.
+
+## Registering to player events
+
+You can listen to the player events by adding an event listener to the player object.
+
+```javascript
+player.addEventListener(player.Event.PLAYER_STATE_CHANGED, e => {
+    const payload = e.payload;
+	// do something with the payload
+    })
+```
+
+## Dispatching player events
+
+In order to dispatch events everywhere on the player we created fake-event.js and fake-event-target.js. These classes were created to provide support dispatching from non-DOM classes.
+
+```javascript
+class YourClass extends FakeEventTarget{
+	someFunction(): void{
+		const data = {};
+		this.dispatchEvent(new FakeEvent(CustomEvents.AUTOPLAY_FAILED, data))
```

Comment:
Wrong.
To dispatch player event you don't need to extend `FakeEventTarget `.
You just need to call the player API:
`player.dispatchEvent(new playkit.core.Event("myEvent", data));
`
Remember that the docs is for **externals** developers.